### PR TITLE
Integrate BUILD.md into the Developer Guide

### PR DESCRIPTION
**Pull request by redhatrises**
- Dedepuplicates all of our docs
- Removes BUILD.md usage for the Developer doc instead
- Removes some legacy content from the Developer Guide

### CODE REVIEW

File: tests/unit/ssg/test_rule_yaml.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import pytest
 
 import os
+import tempfile
```
Comment (shawndwells) [inline tests/unit/ssg/test_rule_yaml.py]: Was this meant for this PR?

File: docs/manual/developer_guide.adoc [hunk 2]
```diff
@@ -8,252 +8,303 @@
...
- - security (intended for fixing security flaws)
- - newpackage (intended for updates introducing new RPM packages)
+To build a package for testing purposes:
 
-options
-Request: select "testing" option of
+```bash
+cd build/
+# disable any product you would not like to bundle in the package. For example:
+cmake -DSSG_PRODUCT_JBOSS_EAP5:BOOL=OFF../
```
[the remark this answers is not in the document]
Comment (redhatrises) [reply docs/manual/developer_guide.adoc]: @yuumasato done

**Comment by redhatrises** [comment]
> > edit: referring to the `Building Official RPMs` section being needed. But on second look lots of that seems outdated. Deferring to @yuumasato about needing release instructions (e.g. bohdi etc). Otherwise this looks good!
> 
> Instructions how to deal with Fedora packaging downstream are linked in [Release Guide](https://github.com/ComplianceAsCode/content/wiki/Release-Guide) step 23.
> 
> I agree with removing github workflow and expectations.
> @redhatrises And what do you think about mentioning https://guides.github.com/ ?

@yuumasato I don't know if we should? Seems out-of-place for our developer guide to me....

**Comment by yuumasato** [comment]
> Seems out-of-place for our developer guide to me....

Well, I agree, but sometimes it is convenient for newcomers. (it can be added later, or somewhere if needed)

Thanks for merging and updating the guides.

**Pull request merged**